sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -23,6 +23,7 @@ import { AbortSignalLike } from "@azure/abort-controller";
 import { EventHubProducer } from "./sender";
 import { EventHubConsumer } from "./receiver";
 import { throwTypeErrorIfParameterMissing } from "./util/error";
+import { isTokenCredential } from "@azure/core-http";
```

Comment:
It feels like an overkill to add a dependency to such a heavy package, for figuring out the type. Let's copy the code into our utils file instead

Reply:
Filed an issue here to discuss moving TokenCredential et al into a more focused library:

https://github.com/Azure/azure-sdk-for-js/issues/3958